apps/emqx_gateway_nats/src/emqx_nats_schema.erl

```diff
@@ -0,0 +1,106 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_nats_schema).
+
+-include("emqx_nats.hrl").
+-include_lib("hocon/include/hoconsc.hrl").
+-include_lib("typerefl/include/types.hrl").
+
+%% config schema provides
+-export([namespace/0, fields/1, desc/1]).
+
+namespace() -> "gateway".
+
+fields(nats) ->
+    [
+        {server_id,
+            sc(binary(), #{
+                desc => ?DESC(server_id),
+                default => <<"emqx_nats_gateway">>
+            })},
+        {server_name,
+            sc(binary(), #{
+                desc => ?DESC(server_name),
+                default => <<"emqx_nats_gateway">>
+            })},
+        {protocol, sc(ref(protocol))},
+        {mountpoint, emqx_gateway_schema:mountpoint()},
+        {listeners, sc(ref(tcp_ws_listeners), #{})}
+    ] ++ emqx_gateway_schema:gateway_common_options();
+fields(protocol) ->
+    [
+        {max_payload_size,
+            sc(
+                non_neg_integer(),
+                #{
+                    default => ?DEFAULT_MAX_PAYLOAD,
+                    desc => ?DESC(max_payload_size)
+                }
+            )}
+    ];
+fields(tcp_ws_listeners) ->
+    [
+        {ws,
+            sc(
+                map(name, ref(ws_listener)),
+                #{
+                    desc => ?DESC(ws_listener)
+                }
+            )},
+        {wss,
+            sc(
+                map(name, ref(wss_listener)),
+                #{desc => ?DESC(wss_listener)}
+            )}
+    ] ++
+        emqx_gateway_schema:fields(tcp_listeners);
+fields(ws_listener) ->
+    [
+        {websocket, sc(ref(websocket), #{})}
```

Comment:
why another level here?
`ws.my_ws_listener1.websocket = {...}`
why not:
`ws.my_ws_listener1 = {...}`

Reply:
The gateway's WebSocket listener follows the same structure as the MQTT WebSocket listener.